spec/services/reporting/agency_and_sp_report_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'csv'
+require 'rails_helper'
+
+RSpec.describe Reporting::AgencyAndSpReport do
+  let(:report_date) do
+    Date.new(2021, 1, 1).in_time_zone('UTC')
+  end
+
+  before { travel_to report_date }
+
+  subject(:report) { described_class.new(report_date) }
+
+  describe '#agency_and_sp_report' do
+
+    subject { report.agency_and_sp_report }
+
+    # OK, so there's existing seed data, it turns out!
+    context 'when adding an IDV SP' do
+      let!(:idv_sp) { create(:service_provider, :idv, :active) }
+
+      let(:expected_report) do
+        [
+          ['', 'Number of apps (SPs)', 'Number of agencies'],
+          ['Auth', 19, 19],
```

Comment:
I expected these to be 0, and my attempt to `let!(:sp_count) { ServiceProvider.count }` results in the wrong number and this is a morning-me problem.